src/main/webapp/script.js

```diff
@@ -92,6 +92,8 @@ function initMap() {
         updateRestaurantList(data.results);
       })
     }))
+
+    gtag('event', 'search', {'search_term': autocomplete.getPlace().name});
```

Comment:
With auto-gps now enabled, you'll have to check whether the location was from User location or not. (if PAC has a place, PAC is used, else user GPS is used)

Reply:
Added separate event fired when GPS updates

Reply:
Sorry i think I wasn't being clear enough.

This gtag event is being fired every time the submit button is being pressed, however there is this scenario whereby the user presses the submit button without choosing a place from PAC.

PAC.getPlace will be null, resulting in an error.

Reply:
Ah I see I get your concern now

`if (!userMarker) {
        // User disabled location and didn't choose a place
        window.alert("Please enter a valid location!");
        return;`
Lines 80-83

If I am not wrong, according to the above lines, if the user doesnt select a valid place, the above check will fire and return out of the function, hence, the tracking event will not be sent. I believe that this should already address that scenario?